eZ/Publish/Core/Repository/Helper/FieldTypeRegistry.php

```diff
@@ -66,18 +63,36 @@ public function getFieldType( $identifier )
             throw new FieldTypeNotFoundException( $identifier );
         }
 
-        if ( $this->fieldTypes[$identifier] instanceof SPIFieldType )
+        $this->initializeFieldType( $identifier );
+
+        return $this->fieldTypes[$identifier];
+    }
+
+    /**
+     * Initializes the FieldType Closure if needed and checks for the correct type
+     *
+     * @throws \RuntimeException If FieldType is not an instance of SPI FieldType interface,
+     *                           or a closure returning the same
+     *
+     * @param string $identifier
+     */
+    protected function initializeFieldType( $identifier )
+    {
+        // First check for Closure and set the return value if found
+        if ( is_callable( $this->fieldTypes[$identifier] ) )
```

Comment:
As it might already be initialized, if we go this way it should instead start with a `if ( $this->fieldTypes[$identifier] instanceof SPIFieldType ) { return; }` clause to avoid further execution and checks.

This is basically why I  preferred original approach as it avoids additional function calls if not needed.
For 6.0 I think we can remove use of closures maybe, I don't think it is widely used anymore, but if it is we might be able to solve it better in compiler passes or something.